python/adios2/stream.py

```diff
@@ -297,6 +297,11 @@ def _(self, name, content, shape=[], start=[], count=[], operations=None):
         if not variable:
             # Sequence variables
             if isinstance(content, np.ndarray):
+                # If shape and count are not specified, use the numpy array's shape
+                if shape == [] and count == []:
+                    shape = list(content.shape)
+                    count = shape
```

Comment:
I think here `count` and `shape` will point to the same memory so if in the future someone changes count, shape will also be updated. Is this something you wanted or you wanted to do `count = list(shape)`?

Reply:
Good catch, thanks @anagainaru . Find the last push fixing this for ndarrays and also for python list by wrapping lists as ndarrays and passing it down to adios2 C/C++ layer.